File: lengthOfLastWord.py
# https://leetcode.com/problems/length-of-last-word/

class Solution(object):
	def lengthOfLastWord(self, s):
		i = len(s) - 1
		while i >= 0 and s[i] == ' ':
			i -= 1

		n = 0
		while i >= 0 and s[i] != ' ':
			i -= 1
			n += 1

		return n

File: lengthOfLastWord.java
// https://leetcode.com/problems/length-of-last-word/

class Solution {
	public int lengthOfLastWord(String s) {
		int i, n;
		for (i = s.length() - 1; i >= 0 && s.charAt(i) == ' '; i--);
		for (i = i, n = 0; i >= 0 && s.charAt(i) != ' '; i--, n++);
		return n;
	}
}
